src/box/errcode.h

```diff
@@ -105,19 +105,39 @@ struct errcode_record {
 	_(ER_TEST_5_ARGS, 10033,		"Test error", "f1", INT, "f2", INT, "f3", INT, "f4", INT, "f5", INT) \
 	_(ER_TEST_FORMAT_MSG, 10050,		"Test error %i %s", "f1", INT, "f2", STRING) \
 	_(ER_TEST_FORMAT_MSG_LESS, 10051,	"Test error %i %s", "f1", INT, "f2", STRING, "f3", INT) \
+	_(ER_TEST_OMIT_TYPE_CHAR, 10060,	"Test error %c", NULL, CHAR) \
+	_(ER_TEST_OMIT_TYPE_INT, 10061,		"Test error %i", NULL, INT) \
+	_(ER_TEST_OMIT_TYPE_UINT, 10062,	"Test error %u", NULL, UINT) \
+	_(ER_TEST_OMIT_TYPE_LONG, 10063,	"Test error %li", NULL, LONG) \
+	_(ER_TEST_OMIT_TYPE_ULONG, 10064,	"Test error %lu", NULL, ULONG) \
+	_(ER_TEST_OMIT_TYPE_LLONG, 10065,	"Test error %lli", NULL, LLONG) \
+	_(ER_TEST_OMIT_TYPE_ULLONG, 10066,	"Test error %llu", NULL, ULLONG) \
+	_(ER_TEST_OMIT_TYPE_STRING, 10067,	"Test error %s", NULL, STRING) \
```

Comment:
An empty string would be easier to read than `NULL` IMO.

Reply:
Fixed.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/errcode.h b/src/box/errcode.h
index 846557541..fc491371d 100644
--- a/src/box/errcode.h
+++ b/src/box/errcode.h
@@ -105,14 +105,14 @@ struct errcode_record {
 	_(ER_TEST_5_ARGS, 10103,		"Test error", "f1", INT, "f2", INT, "f3", INT, "f4", INT, "f5", INT) \
 	_(ER_TEST_FORMAT_MSG, 10200,		"Test error %i %s", "f1", INT, "f2", STRING) \
 	_(ER_TEST_FORMAT_MSG_FEWER, 10201,	"Test error %i %s", "f1", INT, "f2", STRING, "f3", INT) \
-	_(ER_TEST_OMIT_TYPE_CHAR, 10300,	"Test error %c", NULL, CHAR) \
-	_(ER_TEST_OMIT_TYPE_INT, 10301,		"Test error %i", NULL, INT) \
-	_(ER_TEST_OMIT_TYPE_UINT, 10302,	"Test error %u", NULL, UINT) \
-	_(ER_TEST_OMIT_TYPE_LONG, 10303,	"Test error %li", NULL, LONG) \
-	_(ER_TEST_OMIT_TYPE_ULONG, 10304,	"Test error %lu", NULL, ULONG) \
-	_(ER_TEST_OMIT_TYPE_LLONG, 10305,	"Test error %lli", NULL, LLONG) \
-	_(ER_TEST_OMIT_TYPE_ULLONG, 10306,	"Test error %llu", NULL, ULLONG) \
-	_(ER_TEST_OMIT_TYPE_STRING, 10307,	"Test error %s", NULL, STRING) \
+	_(ER_TEST_OMIT_TYPE_CHAR, 10300,	"Test error %c", "", CHAR) \
+	_(ER_TEST_OMIT_TYPE_INT, 10301,		"Test error %i", "", INT) \
+	_(ER_TEST_OMIT_TYPE_UINT, 10302,	"Test error %u", "", UINT) \
+	_(ER_TEST_OMIT_TYPE_LONG, 10303,	"Test error %li", "", LONG) \
+	_(ER_TEST_OMIT_TYPE_ULONG, 10304,	"Test error %lu", "", ULONG) \
+	_(ER_TEST_OMIT_TYPE_LLONG, 10305,	"Test error %lli", "", LLONG) \
+	_(ER_TEST_OMIT_TYPE_ULLONG, 10306,	"Test error %llu", "", ULLONG) \
+	_(ER_TEST_OMIT_TYPE_STRING, 10307,	"Test error %s", "", STRING) \
 
 #else
 #define TEST_ERROR_CODES(_)
@@ -134,7 +134,7 @@ struct errcode_record {
  *
  * In payload fields <NAME> is a string literal and <TYPE> is from
  * enum errcode_field_type. Yet to be consise use short defines like
- * CHAR etc instead of enum. If <NAME> is NULL then the that positional
+ * CHAR etc instead of enum. If <NAME> is "" then the that positional
  * argument will be passed to formatted string but will not be added as
  * error payload.
  */
diff --git a/src/box/error.cc b/src/box/error.cc
index d4a6793ac..bb8d0f58f 100644
--- a/src/box/error.cc
+++ b/src/box/error.cc
@@ -173,52 +173,53 @@ client_error_create(struct error *e, va_list ap)
 	va_end(ap_copy);
 	for (int i = 0; i < r->errfields_count; i++) {
 		const char *name = r->errfields[i].name;
+		bool set_payload = name[0] != '\0';
 		switch (r->errfields[i].type) {
 		case ERRCODE_FIELD_TYPE_CHAR: {
 			char buf[2] = {(char)va_arg(ap, int), '\0'};
-			if (name != NULL)
+			if (set_payload)
 				error_set_str(e, name, buf);
 			break;
 		}
 		case ERRCODE_FIELD_TYPE_INT: {
 			int v = va_arg(ap, int);
-			if (name != NULL)
+			if (set_payload)
 				error_set_int(e, name, v);
 			break;
 		}
 		case ERRCODE_FIELD_TYPE_UINT: {
 			unsigned v = va_arg(ap, unsigned);
-			if (name != NULL)
+			if (set_payload)
 				error_set_uint(e, name, v);
 			break;
 		}
 		case ERRCODE_FIELD_TYPE_LONG: {
 			long v = va_arg(ap, long);
-			if (name != NULL)
+			if (set_payload)
 				error_set_int(e, name, v);
 			break;
 		}
 		case ERRCODE_FIELD_TYPE_ULONG: {
 			unsigned long v = va_arg(ap, unsigned long);
-			if (name != NULL)
+			if (set_payload)
 				error_set_uint(e, name, v);
 			break;
 		}
 		case ERRCODE_FIELD_TYPE_LLONG: {
 			long long v = va_arg(ap, long long);
-			if (name != NULL)
+			if (set_payload)
 				error_set_int(e, name, v);
 			break;
 		}
 		case ERRCODE_FIELD_TYPE_ULLONG: {
 			unsigned long long v = va_arg(ap, unsigned long long);
-			if (name != NULL)
+			if (set_payload)
 				error_set_uint(e, name, v);
 			break;
 		}
 		case ERRCODE_FIELD_TYPE_STRING: {
 			const char *s = va_arg(ap, const char *);
-			if (name != NULL)
+			if (set_payload)
 				error_set_str(e, name, s);
 			break;
 		}
diff --git a/src/box/lua/error.cc b/src/box/lua/error.cc
index cfc740fdb..56fef2abe 100644
--- a/src/box/lua/error.cc
+++ b/src/box/lua/error.cc
@@ -248,7 +248,7 @@ luaT_error_create(lua_State *L, int top_base)
 		     i <= top && argidx < record->errfields_count;
 		     i++, argidx++) {
 			const char *name = record->errfields[argidx].name;
-			if (name != NULL)
+			if (name[0] != '\0')
 				luaT_error_payload_set(L, error, name, i);
 		}
 	}
diff --git a/test/box-luatest/error_subsystem_improvements_test.lua b/test/box-luatest/error_subsystem_improvements_test.lua
index 473b74a49..ff4fca712 100644
--- a/test/box-luatest/error_subsystem_improvements_test.lua
+++ b/test/box-luatest/error_subsystem_improvements_test.lua
@@ -285,7 +285,7 @@ g.test_client_error_creation = function()
     t.assert_equals(e.f2, 'five')
     t.assert_equals(e.f3, 3)
 
-    -- Test if field name is NULL then respective positional argument
+    -- Test if field name is "" then respective positional argument
     -- is printed in formatted string message but not become payload.
     --
     -- We don't support creation of errors with 'l' modifiers in format string.
diff --git a/test/unit/error.c b/test/unit/error.c
index 79d3fc8c3..5a8d65f14 100644
--- a/test/unit/error.c
+++ b/test/unit/error.c
@@ -788,7 +788,7 @@ test_client_error_creation(void)
 	ok(e->errmsg != NULL && strcmp(e->errmsg, "Test error 1 seven") == 0);
 
 	/*
-	 * Test if field name is NULL then respective positional argument
+	 * Test if field name is "" then respective positional argument
 	 * is printed in formatted string message but not become payload.
 	 */
 	diag_set(ClientError, ER_TEST_OMIT_TYPE_CHAR, 'x');

```

</details>